habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_sensors.py

```diff
@@ -176,10 +183,21 @@ def reset_metric(self, *args, episode, task, observations, **kwargs):
         )
 
     def _get_cur_geo_dist(self, task):
-        return np.linalg.norm(
-            np.array(self._sim.robot.base_pos)[[0, 2]]
-            - task.nav_goal_pos[[0, 2]]
+        if len(task.nav_goal_pos.shape) == 1:
+            goals = np.expand_dims(task.nav_goal_pos, axis=0)
+        else:
+            goals = task.nav_goal_pos
+        distance_to_target = np.min(
```

Comment:
In my experience this is pretty slow because there are a large amount of possible goal in objectnav. See if what they are doing [here](https://github.com/facebookresearch/habitat-lab/blob/0ebe5b2949308bf422a3f6179a8af6a94b5e979f/habitat-lab/habitat/tasks/nav/nav.py#L987) also works for you. AFAIK, you can pass multiple goals to the geodesic distance function. Not sure why they are also passing the episode as input to that function though.

Reply:
Yes, I tried passing multiple goals but as far as I remember, it returns `inf` if one of the goals is not reachable (eg. say if the individual geodesic distances are {`inf`, 1, 2, 3}). But I need to check this again.

Reply:
[Here's the source](https://github.com/facebookresearch/habitat-lab/blob/object_rearrange/habitat-lab/habitat/sims/habitat_simulator/habitat_simulator.py#L454) for the geodesic function. It looks like they're passing the episode as a cache if we want to do multiple source queries to the same destination(s). We should probably pass in the episode as well since our set of goals will be the same for all queries.

Reply:
> Yes, I tried passing multiple goals but as far as I remember, it returns inf if one of the goals is not reachable (eg. say if the individual geodesic distances are {inf, 1, 2, 3}). But I need to check this again.

If one of the goal is not reachable, it returns `inf` at times (see below example). I still changed this to pass multiple goals to ensure things aren't slow. We now need to make sure in our episodes that we include only the goals that are reachable from the episode's  `start_position`.

```
(Pdb) self._sim.geodesic_distance(np.array([0, 0, 0]), np.array([[1, 1,  1]]))
inf
(Pdb) self._sim.geodesic_distance(np.array([0, 0, 0]), np.array([[1, 1,  1], [-3.1262, 1.21164, 3.06642]]))
8.522515296936035
(Pdb) self._sim.geodesic_distance(np.array([0, 0, 0]), np.array([[1, 1,  1], [-3.1262, 1.21164, 3.06642], [100, 100, 100]]))
inf
```

Reply:
Is this some non-deterministic behavior. Can you flag this to @aclegg3?

Reply:
It seemed deterministic (a fixed list of goals always gives the same output). But I didn't get a chance to understand the behavior. Yes, I will flag this.

Reply:
Alex fixed this here: https://github.com/facebookresearch/habitat-sim/pull/2014